_posts/2021-04-26-changelog-146.markdown

```diff
@@ -0,0 +1,187 @@
+  Don’t forget to reload a project structure after enabling the corresponding features via `Refresh Cargo Projects`
+  action on Cargo tool window. See [tracking issue](https://github.com/intellij-rust/intellij-rust/issues/6908) for
+  more details and the current status of procedural macros support
+
+  {% include gif-img-wh.html path="/assets/posts/changelog-146/custom-derive" w="600px" h="300px" %}
+
+* [#7093] Now the plugin takes [Cargo config] into account during evaluation of [`cfg` conditions][cfg].
+  Note, it works since Rust 1.52.0 (currently in beta)
+
+  {% include gif-img-wh.html path="/assets/posts/changelog-146/cfg-cargo-config" w="600px" h="300px" %}
+
+* Introduce new intention actions:
+
+  * [#6845], [#7115] `Enable/Disable feature state` to toggles feature state from a `cfg` attribute (by [@Kobzol])
+
+    {% include gif-img-wh.html path="/assets/posts/changelog-146/intention-cargo-feature" w="600px" h="300px" %}
+    By the way, we have a [blog post][cargo features blogpost] about cargo features support
+
+  * [#7071] `Share in Playground` to share a selected Rust code on [Rust Playground](https://play.rust-lang.org)
+
+    {% include gif-img.html path="/assets/posts/changelog-146/intention-playground" w="600px" %}
+    Previously, this feature was only available as `Rust | Share in Playground` action in `Tools` and context menus
+
+  * [#6960] `Expand dependency specification` for `Cargo.toml`
+
+    {% include gif-img-wh.html path="/assets/posts/changelog-146/intention-expand-dependency" w="600px" h="300px" %}
+
+  * [#6837] `Create struct` to create a struct from an unresolved struct literal (by [@Kobzol])
+
+    {% include gif-img-wh.html path="/assets/posts/changelog-146/intention-struct" w="600px" h="300px" %}
+
+* [#5650] Improve `Use destructuring declaration` intention action: replace usages of destructured variable (by [@Kobzol])
+
+  {% include gif-img-wh.html path="/assets/posts/changelog-146/intention-destructure-decl" w="600px" h="150px" %}
+
+* [#6986] Allow specifying the function return type in `Create function` intention action if the inferred return type is unknown (by [@Kobzol])
+
+  {% include gif-img-wh.html path="/assets/posts/changelog-146/intention-create-function" w="600px" h="300px" %}
+
+* Improve [code folding]:
+
+  * [#7111] Support folding for multiline `where` clauses (by [@Stzx])
+
+    {% include gif-img-wh.html path="/assets/posts/changelog-146/fold-where" w="600px" h="300px" %}
+
+  * [#7112] Enable folding for series of use items by default (by [@Stzx])<br>
+    You can disable it via `Imports` option in `Settings | Editor | General | Code Folding` settings
+
+  * [#7112] Support `Method bodies` option of `Settings | Editor | General | Code Folding - General` settings
+    to fold function/method bodies by default (by [@Stzx])
+
+* Improve experimental crate-local index:
```

Comment:
Let's add a link to https://github.com/intellij-rust/intellij-rust/issues/6463.
I've asked @avrong to add a paragraph with description of crate local index feature